### DISCUSSION TITLE
How to prevent reloading page when using `pagy_items_selector_js ` with TurboRails

### DISCUSSION THREAD

**Discussion by nnhutan** [ddnexus/pagy#681] [Feature Requests]
I'm currently using TurboRails for my Rails app. I use `pagy_items_selector_js` helper. I found that when we submit the number of items in the input the script will update `window.location.href` so it causes the reloading page. In this case, if I use `Turbo.visit` or the `a` tag to navigate, the page does not reload.
[related codes](https://github.com/ddnexus/pagy/blob/20e230d7a339fe2510f9a60b9157210e4109efed/gem/javascripts/pagy-module.js#L56-L71):
```js
        const action = function () {
            if (input.value === initial) {
                return;
            } // not changed
            const [min, val, max] = [input.min, input.value, input.max].map(n => parseInt(n) || 0);
            if (val < min || val > max) { // reset invalid/out-of-range
                input.value = initial;
                input.select();
                return;
            }
            let [page, url] = getVars(input.value); // eslint-disable-line prefer-const
            if (typeof trimParam === "string" && page === "1") {
                url = trim(url, trimParam);
            }
            window.location.href = url;
        };
```

**Comment by ddnexus** [comment]
@nnhutan yes, it looks like turbo intercepts only regular `a` clicks, so it misses that `location` change ATM.

I will add an extra keyword argument to the ruby helper that will allow you to customize the helper behavior (per helper) with whatever javascript statement you prefer to handle the `url`:

```rb
pagy_items_selector_js(@pagy, goto_url_js: 'Turbo.visit(url);')
```
It will work the same also for `pagy_combo_nav_js` that has the same problem.

Let me see if I can release it in a couple of days  😎

**Reply by ddnexus** [reply]
The alternative would be using the classic hack of the `a` element click, but I removed it from a previous version because it was ugly. So either I will find a cleaner way to do it, or I will use the extra argument. 🤔

**Comment by ddnexus** [comment]
Implemented in [8.0.2](https://github.com/ddnexus/pagy/releases/tag/8.0.2).
_(It should just work without any argument)_
